src/hooks/useAlternateLinks.ts

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { getBasePath } from '../helpers/getBasePath';
+import { getOrigin } from '../helpers/getOrigin';
+import { getSourceUrl } from '../helpers/getSourceUrl';
+import { useRewrites } from './useRewrites';
+
+interface AlterNateLinks {
+  href: string;
+  hrefLang: string;
+}
+
+export const useAlternateLinks = (currentLocale: string): AlterNateLinks[] => {
+  const { basePath, pathname, locales } = useRouter();
+  const rewrites = useRewrites();
+  const [alternateLinks, setAlternateLinks] = useState<AlterNateLinks[]>([]);
+
+  useEffect(() => {
+    const origin = getOrigin();
+    const alternateLinks = locales.map((lang) => {
+      const alternateLink = getSourceUrl({ rewrites, locale: lang, path: pathname });
+      return {
+        href: `${origin}${getBasePath(basePath)}${lang}${alternateLink}`,
+        hrefLang: lang,
+      };
+    });
+    setAlternateLinks(
+      pathname !== '/'
+        ? alternateLinks
+        : [...alternateLinks, { href: `${origin}${getBasePath(basePath)}`, hrefLang: 'x-default' }]
```

Comment:
Hi @eemeli,

Why an argument or prop?

If I place the `<IntlHead />` component in `<Layout />` component, the idea would be to not have to care about whether or not I'm in the home page and let the component handle this.

Of course, I can always pass a prop to `<Layout />` and then to `<IntlHead />` but I'm failing to understand the benefit.

Can you please elaborate on this one.

Thanks

Reply:
While using the root page as a language switch page is relatively common for sites that implement it, it's by no means always the desired thing. For instance, it may be desirable not to have any `x-default` target, or it may be desirable to use a page at a different path for this purpose.

Reply:
I'm not sure sur about this. According to the requirements provided by @nbouvrette, `x-default` should only go into pathname `/`whether we have a basePath or not.

Also, according to the best SEO practices, `x-default` should only go in the root of the site no?

I can definitely add a prop but I don't think this is the desired behaviour. @nbouvrette What do you think?

Reply:
`x-default` should go to any URL (alternate) that has a version "without a predefined locale" and that includes an auto-detection mechanism. 

Normally only the homepage should have this "auto-detection" behavior. But depending on our implementation (e.g. having a "catch-all" locale) we might have alternative URLs without locale prefix available. Unless they are doing auto-detection (which would require server-side checks for all pages and impact performance) I think we should avoid using `x-default` outside the homepage - not sure if I understood the entire situation correctly on the other hand :)

Reply:
Next.js will get the locale from the URL, whether we have `localeDetection` set to true or false, if the locale is one in the supported locales it will automatically set the locale as the locale in the URL.

`localeDetection` is intended to work in the root of the site `/` if we arrive at `/en-CA` then the locale will be set to `en-CA` same thing for any other supported locale.

Now, Next.js, out of the box, will not add the default locale to the URL, so if we arrive at `/about-us` it will not add the default locale. However we don't want this right, we want to inject the default locale so `/about-us` will automatically redirect to /en-CA/about-us, thus we should never have an URL without a locale.

Knowing this, do we want to have `x-default` elsewhere other than at the root of the site `/`? Note that the root has nothing to do with wether we have defined a `basePath` or not. If we defined a `basePath` in `next.config.js `file, the pathname will still be `/`